### PR TITLE
Initial authentication backend

### PR DESCRIPTION
**Pull request by danigm**
https://phabricator.endlessm.com/T32614

### CODE REVIEW

**Review by dylanmccall — CHANGES_REQUESTED** [review]
This looks like a good start. Thanks for getting things moving, @danigm! I agree with using a login URL rather than a middleware thing. That should be nice and easy to work with.

So, just one small request about naming the thing :)

File: Pipfile [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+[packages]
+flake8 = "==3.8.3"
+pre-commit = "==1.15.1"
+twine = "*"
+
+[dev-packages]
+
+[requires]
+python_version = "3.7"
```
[the remark this answers is not in the document]
Comment (danigm) [reply Pipfile]: I've added the lock file

File: kolibri_desktop_auth_plugin/kolibri_plugin.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+from . import settings
+
+
+class DaemonAuth(KolibriPluginBase):
+    translated_view_urls = "urls"
+    settings_module = settings
+
+    @property
+    def url_slug(self):
+        return r"daemon-auth/"
```
Comment (manuq) [inline kolibri_desktop_auth_plugin/kolibri_plugin.py:12]: The `deamon-auth/` URL prefix doesn't say much to me. I'd prefer if it includes something about "desktop user" or "system user". What about?: 
```suggestion
        return r"desktop-user-auth/"
```

Or maybe even better, leave it to the default which is a lower-case version of the plugin name. And rename the class name to DesktopAuth or DesktopUserAuth (I'm tempted to request a repo rename again to `kolibri-desktop-user-auth-plugin`).

File: kolibri_desktop_auth_plugin/migrations/0001_initial.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+    initial = True
+
+    dependencies = [
+        migrations.swappable_dependency(settings.AUTH_USER_MODEL),
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name="DaemonAuthUser",
```
Comment (manuq) [inline kolibri_desktop_auth_plugin/migrations/0001_initial.py:21]: This should be renamed also along with the other renames for consistency. Maybe "DesktopUser"?

**Pull request merged**